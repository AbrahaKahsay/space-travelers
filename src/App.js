import './App.css';
import { Routes, Route } from 'react-router-dom';
import Navbar from './components/Navbar';
import Mission from './components/Mission/Mission';
import Profile from './components/Profile/Profile'
import Rockets from './components/Rockets/Rockets';

function App() {
  return (
    <>
      <Navbar />
      <Routes>
        <Route path="/" element={<Rockets />} />
        <Route path="/mission" element={<Mission />} />
        <Route path="/profile" element={<Profile />} />
      </Routes>
    </>

  );
}

export default App;