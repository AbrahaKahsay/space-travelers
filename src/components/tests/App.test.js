import renderer from 'react-test-renderer';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../Redux/configureStore';
import App from '../../App';

describe('Body of the whole Rocket Page', () => {
  it('Check Falcon 1 in the UI', async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    );
    await waitFor(() => {
      const waitForDOM = screen.findByText(
        "The Falcon 1 was an expendable"
      );
      expect(waitForDOM).not.toBeNull();
    });
  });

  it('It should matchsnapshot', () => {
    const Tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(Tree).toMatchSnapshot();
  });
});
