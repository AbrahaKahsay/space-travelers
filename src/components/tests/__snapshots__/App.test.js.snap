// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body of the whole Rocket Page It should matchsnapshot 1`] = `
Array [
  <nav
    className="navbar"
  >
    <div
      className="logo-container"
    >
      <div>
        <img
          alt="Logo"
          className="logo"
          src="planet.png"
        />
      </div>
      <h1>
        Space Travelers' Hub
         
      </h1>
    </div>
    <div
      className="links"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="link-list active"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className="link-list"
            href="/mission"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li>
          <a
            className="link-list"
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div />,
]
`;
